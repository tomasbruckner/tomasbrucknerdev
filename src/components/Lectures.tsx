import React, { FC } from 'react';
import { t } from '../utils/i18n';
import VideoCard from './VideoCard';
import { makeStyles, Typography } from '@material-ui/core';

const Videos = Object.freeze({
  GarbageCollector: {
    title: 'videos.garbagecollector',
    href: 'https://www.youtube.com/embed/MuHCvf7WZys',
  },
  Microfrontend: {
    title: 'videos.microfrontend',
    href: 'https://www.youtube.com/embed/kmeFVANlHmU',
  },
  Microservices: {
    title: 'videos.microservices',
    href: 'https://www.youtube.com/embed/04-iN3DdIKQ',
  },
  Neuron: { title: 'videos.neuron', href: 'https://www.youtube.com/embed/vXn3JDmjzVE' },
  Babel: { title: 'videos.babel', href: 'https://www.youtube.com/embed/Me82U4c6K3s' },
  Aws: { title: 'videos.aws', href: 'https://www.youtube.com/embed/rIDNZrXMbIo' },
  Alexa: { title: 'videos.alexa', href: 'https://www.youtube.com/embed/Ueu4tOIse4I' },
  Mongo2: { title: 'videos.mongo2', href: 'https://www.youtube.com/embed/Mil2i6erarY' },
  Mongo1: { title: 'videos.mongo1', href: 'https://www.youtube.com/embed/rDk_T_F34-w' },
  Docker: { title: 'videos.docker', href: 'https://www.youtube.com/embed/h2kA2RlLzOs' },
  Adonis: { title: 'videos.adonis', href: 'https://www.youtube.com/embed/wrfG8fhm5fE' },
  Unfurling: { title: 'videos.unfurling', href: 'https://www.youtube.com/embed/6zkWZg2-bEI' },
} as const);

const useStyles = makeStyles((theme) => ({
  text: {
    marginTop: theme.spacing(2),
  },
}));

const Lectures: FC = () => {
  const classes = useStyles();

  return (
    <>
      <Typography variant="h4" id="lectures" gutterBottom className={classes.text}>
        {t('navigation.lectures')}
      </Typography>
      {Object.keys(Videos).map((v) => {
        const { title, href } = Videos[v];

        return <VideoCard videoUrl={href} videoHeadline={t(title)} key={title} />;
      })}
    </>
  );
};

export default Lectures;
