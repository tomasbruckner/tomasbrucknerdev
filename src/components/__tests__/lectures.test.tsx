import React from 'react';
import Lectures from '../Lectures';
import renderer from 'react-test-renderer';

it('Lectures renders correctly', () => {
  const tree = renderer.create(<Lectures />).toJSON();
  expect(tree).toMatchSnapshot();
});
