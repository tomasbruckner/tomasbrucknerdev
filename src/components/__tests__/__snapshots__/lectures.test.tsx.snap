// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lectures renders correctly 1`] = `
Array [
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
    id="lectures"
    style={
      Object {
        "marginTop": "1em",
      }
    }
  >
    Lekce
  </h4>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        Microfrontendy
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/kmeFVANlHmU"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="Microfrontendy"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        Microservisy
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/04-iN3DdIKQ"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="Microservisy"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        Neuronové sítě
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/vXn3JDmjzVE"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="Neuronové sítě"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        Babel - Jak si udělat vlastní plugin
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/Me82U4c6K3s"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="Babel - Jak si udělat vlastní plugin"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        AWS - To nejzajímavější
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/rIDNZrXMbIo"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="AWS - To nejzajímavější"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        Alexa - Jak si udělat vlastní aplikaci
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/Ueu4tOIse4I"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="Alexa - Jak si udělat vlastní aplikaci"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        MongoDB - Část 2
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/Mil2i6erarY"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="MongoDB - Část 2"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        MongoDB - Část 1
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/rDk_T_F34-w"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="MongoDB - Část 1"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        Docker - Performance problém na Windows a OSX a jak ho řešit
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/h2kA2RlLzOs"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="Docker - Performance problém na Windows a OSX a jak ho řešit"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        AdonisJS - Případová studie
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/wrfG8fhm5fE"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="AdonisJS - Případová studie"
        width="100%"
      />
    </div>
  </div>,
  <div
    className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    style={
      Object {
        "marginBottom": "48px",
        "marginTop": "48px",
        "minWidth": 275,
      }
    }
  >
    <div
      className="MuiCardContent-root"
    >
      <h2
        className="MuiTypography-root MuiTypography-h5"
        style={
          Object {
            "marginBottom": "16px",
          }
        }
      >
        Unfurling - Jak definovat preview webové stránky
      </h2>
      <iframe
        height="440"
        src="https://www.youtube.com/embed/6zkWZg2-bEI"
        style={
          Object {
            "borderWidth": 0,
          }
        }
        title="Unfurling - Jak definovat preview webové stránky"
        width="100%"
      />
    </div>
  </div>,
]
`;
